sxg_rs/src/process_html.rs

```diff
@@ -0,0 +1,241 @@
+pub fn process_html(input: HttpResponse, option: ProcessHtmlOption) -> Result<HttpResponse> {
+    let content_type_header = input.headers.iter().find_map(|(name, value)| {
+        if name.eq_ignore_ascii_case("content-type") {
+            Some(value)
+        } else {
+            None
+        }
+    });
+    let mut known_utf8 = false;
+    if let Some(content_type_header) = content_type_header {
+        match parse_content_type(content_type_header) {
+            ContentType::HtmlUtf8 => known_utf8 = true,
+            ContentType::HtmlOther => (),
+            ContentType::Other => return Ok(input),
+        };
+    } else {
+        // Doesn't process HTML because content-type header does not exsist.
+        return Ok(input);
+    }
+    let input_body = match String::from_utf8(input.body) {
+        Ok(input_body) => input_body,
+        Err(e) => {
+            // Doesn't process HTML because input body bytes can't be parsed at UTF-8 string, for
+            // example, a UTF-16 BOM exsists.
+            return Ok(HttpResponse {
+                body: e.into_bytes(),
+                headers: input.headers,
+                status: input.status,
+            });
+        }
+    };
+    let mut link_headers: Vec<String> = vec![];
+    let element_content_handlers = vec![
+        // Parse the meta tag, per the implementation in HTMLMetaCharsetParser::CheckForMetaCharset:
+        // https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/parser/html_meta_charset_parser.cc;l=62-125;drc=7a0b88f6d5c015fd3c280b58c7a99d8e1dca28ac.
+        // This differs slightly from what's described at https://github.com/whatwg/html/issues/6962, and
+        // differs drastically from what's specified in
+        // https://html.spec.whatwg.org/multipage/parsing.html#prescan-a-byte-stream-to-determine-its-encoding.
+        lol_html::element!("meta", |e| {
+            // EncodingFromMetaAttributes:
+            // https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/parser/html_parser_idioms.cc;l=362-393;drc=7a0b88f6d5c015fd3c280b58c7a99d8e1dca28ac
+            if let Some(charset) = e.get_attribute("charset") {
+                if charset.eq_ignore_ascii_case("utf-8") {
+                    known_utf8 = true;
+                }
+            } else if let (Some(http_equiv), Some(content)) =
+                (e.get_attribute("http-equiv"), e.get_attribute("content"))
+            {
+                if http_equiv.eq_ignore_ascii_case("content-type")
+                    && parse_content_type(&content) == ContentType::HtmlUtf8
+                {
+                    known_utf8 = true;
+                }
+            }
+            Ok(())
+        }),
+        lol_html::element!(
+            r#"link[rel~="preload" i][href][as]:not([data-sxg-no-header])"#,
+            |e| {
+                if let Some(href) = e.get_attribute("href") {
```

Comment:
Maybe send a follow-up to delete `as?.match(TOKEN)` from processor.ts. I believe that it no longer necessary because:
- We escape attribute values now.
- We collect all attributes now. (Before landing `data-sxg-no-header`, I had suggested using something like `as=imag&x65;` to opt out of processing, because we didn't support `crossorigin` at the time.)

Reply:
Adding a TODO